tests/unit/lms/services/canvas_files_test.py

```diff
@@ -0,0 +1,57 @@
+from unittest.mock import sentinel
+
+import pytest
+
+from lms.models import CanvasFile
+from lms.services.canvas_files import factory
+from tests import factories
+
+
+class TestUpsert:
+    def test_if_theres_no_record_in_the_db_it_adds_one(self, db_session, svc):
+        canvas_file = factories.CanvasFile.build()
```

Comment:
This creates a `CanvasFile` object _without_ adding it to the DB session (factory boy's `build()` method)